ci/provisioning/post_provision_config_common_functions.sh

```diff
@@ -0,0 +1,188 @@
+#!/bin/bash
+
+set -eux
+
+: "${DAOS_STACK_RETRY_DELAY_SECONDS:=60}"
+: "${DAOS_STACK_RETRY_COUNT:=3}"
+: "${DAOS_STACK_MONITOR_SECONDS:=600}"
+: "${BUILD_URL:=Not_in_jenkins}"
+: "${STAGE_NAME:=Unknown_Stage}"
+: "${OPERATIONS_EMAIL:=$USER@localhost}"
+
+retry_dnf() {
+    local monitor_threshold="$1"
+    shift
+
+    local args=("dnf" "-y" "${@}")
+    local attempt=0
+    local rc=0
+    while [ $attempt -lt "${RETRY_COUNT:-$DAOS_STACK_RETRY_COUNT}" ]; do
+        if monitor_cmd "$monitor_threshold" "${args[@]}"; then
+            # Command succeeded, return with success
+            if [ $attempt -gt 0 ]; then
+                send_mail "Command retry successful in $STAGE_NAME after $attempt attempts using ${repo_server:-nexus} as initial repo server " \
+                          "Command:  ${args[*]}\nAttempts: $attempt\nStatus:   $rc"
+            fi
+            return 0
+        fi
+        # Command failed, retry
+        rc=${PIPESTATUS[0]}
+        (( attempt++ )) || true
+        if [ "$attempt" -gt 0 ]; then
+            if [ "$attempt" -eq 2 ] && [ ${#repo_servers[@]} -gt 1 ]; then
```

Comment:
(lint) repo_servers is referenced but not assigned (did you mean 'repo_server'?). [SC2154]